Utils/SceneUtils.cs

```diff
@@ -250,6 +251,67 @@ public static void GetActors()
             }
         }
 
+        public static void UpdateSceneByNumber(int SceneNumber)
+        {
+            for (int i = Math.Min(RomData.SceneList.Count, SceneNumber); i >= 0; i--)
+            {
+                if (RomData.SceneList[i].Number == SceneNumber)
+                {
+                    UpdateScene(RomData.SceneList[i]);
+                }
+            }
+        }
+
+        public static void WriteShuffledDungeonChests(int[][] ChestShuffle)
+        {
+            SceneUtils.ReadSceneTable();
+            SceneUtils.GetMaps();
+            SceneUtils.GetMapHeaders();
+            SceneUtils.GetActors();
+            // WF, SH, GB, ST, IST
+            int[] DungeonScene = new int[] { 0x1B, 0x21, 0x49, 0x16, 0x18 };
+            // WF
+            // spinning flower fairy, first room fairy, Compass, Small Key, Map, Bow, BK, dark room fairy
+            // SH
+            // Compass Fairy, Compass, Freezard Bridge Key, Push block fairy, push block key
+            // Lava Fairy, BK, Deku Flower Fairy, Freezard Stalagmite Fairy, Fire Arrows
+            // Ice Stalactite Fairy, Ice Stalactite Key, Invisible Stair Platform Fairy, Map
+            // GB
+            // Map, Bio Baba Fairy, BK, Key, Compass, Ice Arrows, Water Wheel Fairy, Hidden Torch Fairy, 
+            // Seesaw Fairy, Yellow/Green Reservoir Fairy, First Room Fairy
+            // ST ... oh god ... I'll suffix inverted temple fairy chests with (i)
+            // Remains Platform (i), Lower First Room, Upper First Room, Eyegore, Eyegore Sunblock,
+            // Giant, Eyegore Key, Elegy Key, Wind Elevator (i), Frozen Eye Switch (i), Underwater Sun Switch [have to fact check if this is the one tied to the sun switch]
+            // Beetle Guarded Key, Compass, Wizzrobe (i), Wizzrobe Key, Wizzrobe, Map [assuming that's the one in the back]
+            // Mirror Shield Sun Switch, Sunblock Room, Post Light Arrows Room, Fire Ring, Rupee Nook, Light Arrows
+            int DungeonSceneNumber;
+            for (int d = 0; d < DungeonScene.Length; d++)
```

Comment:
`DungeonScene` should be plural `dungeonScenes`, since it's a list. This line can also be simplified with a foreach loop: `foreach(int dungeonId in dungeonScenes)`

Reply:
I needed the index later to index in the Chest Shuffle parameter, it was a foreach loop beforehand. Though it might be better to just keep track of that index separately now that you mention it.